### PR TITLE
middleware/kubernetes: Rewrite/expand readme

### PR DESCRIPTION
**Pull request by chrisohaver**
* Match formatting and style to other middleware READMEs.
* Expand the autopath 
* Add examples section

I'm using a new visual markdown editor (macdown)... so I dont know how compatible it _really_ is with github yet.  Will see - I'll tweak it if things look off.

### CODE REVIEW

File: middleware/kubernetes/README.md [hunk 1]
```diff
@@ -1,204 +1,250 @@
 # kubernetes
 
-*kubernetes* enables reading zone data from a kubernetes cluster.
-It implements the [spec](https://github.com/kubernetes/dns/blob/master/docs/specification.md)
-defined for kubernetes DNS-Based service discovery:
+The *kubernetes* middleware enables the reading zone data from a Kubernetes cluster.  It implements the [Kubernetes DNS-Based Service Discovery Specification](https://github.com/kubernetes/dns/blob/master/docs/specification.md).
 
-Service `A` records are constructed as "myservice.mynamespace.svc.coredns.local" where:
+## Syntax
```
Comment (johnbelamaric) [inline middleware/kubernetes/README.md]: Make a note that everything except ZONE is optional.

**Comment by chrisohaver** [comment]
Only rendering issue I see is that the section breaks are being rendered as bulllets.
I'll remove them.

File: middleware/kubernetes/README.md [hunk 2]
```diff
@@ -1,204 +1,250 @@
...
+
+	pods POD-MODE
+
+Set the mode for handling IP-based pod A records, e.g. `1-2-3-4.ns.pod.cluster.local. in A 1.2.3.4`.  This option is provided to facilitate use of SSL certs when connecting directly to pods.
+
+Valid values for **POD-MODE**:
+
+* `disabled`: Default. Do not process pod requests, always returning `NXDOMAIN`
+
+* `insecure`: Always return an A record with IP from request (without checking k8s).  This option is is vulnerable to abuse if used maliciously in conjunction with wildcard SSL certs.
```
Comment (johnbelamaric) [inline middleware/kubernetes/README.md]: It is provided for backward compatibility to Kube-DNS.

File: middleware/kubernetes/README.md [hunk 3]
```diff
@@ -1,204 +1,250 @@
...
+-
+### cidrs
+
+	cidrs CIDR [CIDR...]
 	
-	# federation <federation-name> <federation-domain>
-	#
-	# Defines federation membership.  One line for each federation membership.
-	# Each line consists of the name of the federation, and the domain.
+Expose cidr ranges to reverse lookups.  Include any number of space delimited cidrs, and or multiple cidrs options on separate lines. The Kubernetes middleware will respond to PTR requests for ip addresses that fall within these ranges.
```
Comment (johnbelamaric) [inline middleware/kubernetes/README.md]: s?and or?and/or?

File: middleware/kubernetes/README.md [hunk 4]
```diff
@@ -1,204 +1,250 @@
...
-	# If a query for a record in the cluster zone results in NXDOMAIN,
-	# normally that is what the response will be. However, if you specify
-	# this option, the query will instead be passed on down the middleware
-	# chain, which can include another middleware to handle the query.
-	fallthrough
-}
+	# host -t a google.com.default.svc.cluster.local.
+	google.com has address 216.58.194.206
+
+**NDOTS** (default: `0`) This provides an adjustable threshold to prevent server side lookups from triggering. If the number of dots before the first search domain is less than this number, then the search path will not executed on the server side.  When autopath is enabled with default settings, the search path is always conducted on the server side.
```
Comment (johnbelamaric) [inline middleware/kubernetes/README.md]: maybe "when auto path is enabled with default settings, the search path is conducted whenever the query ends with `<namespace>.svc.<zone>`"

File: middleware/kubernetes/README.md [hunk 5]
```diff
@@ -1,204 +1,250 @@
...
+
+**RESOLV-CONF** (default: `/etc/resolv.conf`) If specified, the kubernetes middleware uses this file to get the host's search domains. The kubernetes middleware performs a lookup on these domains if the in-cluster search domains in the path fail to produce an answer. If not specified, the values will be read from the local resolv.conf file (i.e the resolv.conf file in the pod containing CoreDNS).  In practice, this option should only need to be used if running CoreDNS outside of the cluster and the search path in /etc/resolv.conf does not match the cluster's "default" dns-policiy.
+
+Enabling autopath requires more memory, since it needs to maintain a watch on all pods. If autopath and "pods verified" mode are both enabled, they will share the same watch. Enabling both options should have an equivalent memory impact of just one.
```
Comment (johnbelamaric) [inline middleware/kubernetes/README.md]: "pods verified" -> `pods verified`

**Review by johnbelamaric — CHANGES_REQUESTED** [review]
nits

File: middleware/kubernetes/README.md [hunk 6]
```diff
@@ -1,204 +1,250 @@
...
-	#
-	cidrs 10.0.0.0/24 10.0.10.0/25
-
-	# upstream <address> [<address>] ...
-	#
-	# Defines upstream resolvers used for resolving services that point to
-	# external hosts (External Services).  <address> can be an ip, an ip:port, or
-	# a path to a file structured like resolv.conf.
-	upstream 12.34.56.78:53
+### tls
```
Comment (miekg) [inline middleware/kubernetes/README.md]: These are all bullets in the other README's and there is an example section where we put examples + explanations.

Comment (chrisohaver) [reply middleware/kubernetes/README.md]: Bullets are ok for brief explanations for simple options.  But many of the k8s options are quite involved, and require lengthy explanations.  I tried fitting that into bullets, but it was making it very hard to read, so I went with separate sections.

I have included an example section with several full examples.

Comment (chrisohaver) [reply middleware/kubernetes/README.md]: Actually - i just figured out how to indent paragraphs and code blocks to match the bullet indentation above it... so i'll change them to bullets

**Comment by chrisohaver** [comment]
github doesn't render the indentation for bullets...

fun.

**Comment by chrisohaver** [comment]
I'll leave the bullets in since its consistent with other middleware READMEs. But IMO it's harder to read.

**Comment by chrisohaver** [comment]
> github doesn't render the indentation for bullets...

OK - I've got the indentation working correctly now.  Much easier to read.

File: middleware/kubernetes/README.md [hunk 7]
```diff
@@ -1,204 +1,264 @@
...
 
-* "myservice" is the name of the k8s service
-* "mynamespace" is the k8s namespace for the service, and
-* "svc" indicates this is a service
-* "coredns.local" is the zone
+## Syntax
+
+```
+kubernetes ZONE [ZONE...] [{
+	[resyncperiod PERIOD]
```
Comment (miekg) [inline middleware/kubernetes/README.md]: Nit: PERIOD -> DURATION (consistent w/ other docs)

File: middleware/kubernetes/README.md [hunk 8]
```diff
@@ -1,204 +1,264 @@
...
+	[endpoint URL
+	[tls CERT-FILE KEY-FILE CACERT-FILE]]
+	[namespaces NAMESPACE [NAMESPACE...]]
+	[labels EXPRESSION [,EXPRESSION...]]
```
Comment (miekg) [inline middleware/kubernetes/README.md]: CIDR is space delimited....

File: middleware/kubernetes/README.md [hunk 9]
```diff
@@ -1,204 +1,264 @@
...
+		resyncperiod 15m
+	}
+  ```
 
-* "epname" is the hostname (or name constructed from IP) of the endpoint
-* "myservice" is the name of the k8s service that the endpoint serves
-* "mynamespace" is the k8s namespace for the service, and
-* "svc" indicates this is a service
-* "coredns.local" is the zone
+* **endpoint URL**
```
Comment (miekg) [inline middleware/kubernetes/README.md]:
~~~
`endpoint`
~~~
for the name
~~~
**URL**
~~~
for the variable your referencing, see middleware.md

File: middleware/kubernetes/README.md [hunk 10]
```diff
@@ -1,204 +1,264 @@
...
+	[cidrs CIDR [CIDR...]]
+	[upstream ADDRESS [ADDRESS...]]
+	[federation NAME DOMAIN]
+	[autopath [NDOTS [RESPONSE [RESOLV-CONF]]]
+	[fallthrough]
+}]
+```
 
-Pod `A` records are constructed as "1-2-3-4.mynamespace.pod.coredns.local" where:
+* *resyncperiod* **DURATION**
```
Comment (miekg) [inline middleware/kubernetes/README.md]:
~~~
`resyncperiod`
~~~
for all these keywords

**Review by miekg — COMMENTED** [review]
nits, otherwise LGTM

**Pull request merged**